tests/src/Mod/Part/App/TopoShapeMakeShapeWithElementMap.cpp

```diff
@@ -48,14 +55,53 @@ class TopoShapeMakeShapeWithElementMapTests: public ::testing::Test
     Part::TopoShape::Mapper _mapper;
 };
 
-TEST_F(TopoShapeMakeShapeWithElementMapTests, nullShapeThrows)
+TEST_F(TopoShapeMakeShapeWithElementMapTests, nullShapesThrows)
 {
     // Arrange
     auto [cube1, cube2] = TopoShapeExpansionHelpers::CreateTwoCubes();
     std::vector<Part::TopoShape> sources {cube1, cube2};
-    TopoDS_Vertex nullShape;
+    TopoDS_Vertex nullVertex;
+    TopoDS_Edge nullEdge;
+    TopoDS_Wire nullWire;
+    TopoDS_Face nullFace;
+    TopoDS_Shell nullShell;
+    TopoDS_Solid nullSolid;
+    TopoDS_CompSolid nullCompSolid;
+    TopoDS_Compound nullCompound;
 
     // Act and assert
-    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullShape, *Mapper(), sources),
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullVertex, *Mapper(), sources),
                  Part::NullShapeException);
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullEdge, *Mapper(), sources),
+                 Part::NullShapeException);
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullWire, *Mapper(), sources),
+                 Part::NullShapeException);
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullFace, *Mapper(), sources),
+                 Part::NullShapeException);
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullShell, *Mapper(), sources),
+                 Part::NullShapeException);
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullSolid, *Mapper(), sources),
+                 Part::NullShapeException);
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullCompSolid, *Mapper(), sources),
+                 Part::NullShapeException);
+    EXPECT_THROW(Shape()->makeShapeWithElementMap(nullCompound, *Mapper(), sources),
+                 Part::NullShapeException);
+}
+
+TEST_F(TopoShapeMakeShapeWithElementMapTests, emptySourceShapes)
+{
+    // Arrange
+    auto [cube1, cube2] = TopoShapeExpansionHelpers::CreateTwoCubes();
+    std::vector<Part::TopoShape> emptySources;
+    std::vector<Part::TopoShape> nonEmptySources {cube1, cube2};
+
+    // Act and assert
+    for (auto& source : nonEmptySources) {
+        Part::TopoShape& modifiedShape = source;
+        Part::TopoShape& originalShape = source;
```

Comment:
Not sure if it makes sense to add `modifiedShape` and `originalShape`, it could be the same of running the test only using `source` as all of them points to the same object

Reply:
It makes sense to me; it clarifies that you're testing to prove makeShapeWithElementMap does nothing in this case.